### PR TITLE
Minor fixes in tekton resources

### PR DESCRIPTION
**Pull request by geetikakay**
##### Short description:
yaml_file, name, namespace, and client can  now be  passed as part of the kwargs to allow more flexibility for tekton resources. Class is now  capable of handling them dynamically. 
## Summary by CodeRabbit

- **New Features**
	- Introduced an optional `yaml_file` parameter to the `Task` class for enhanced configuration.

- **Bug Fixes**
	- Improved error handling in the `to_dict` method across `Pipeline`, `Task`, and `PipelineRun` classes to ensure necessary attributes are validated correctly when `yaml_file` is not provided.
	- Streamlined the initialization process of the `Pipeline` and `PipelineRun` classes by removing redundant parameters.

### CODE REVIEW

File: ocp_resources/pipeline.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def __init__(
             name=name,
             namespace=namespace,
             client=client,
+            yaml_file=yaml_file,
```
Comment (myakove) [inline ocp_resources/pipeline.py:38]: `yaml_file` must be passed to `super`, also `name`, `namespace` and `client`needs to be removed and be part of `kwargs`

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
Please add to the commit msg why this change is needed and what thus solved 

Thanks

File: ocp_resources/pipeline.py [hunk 2]
```diff
@@ -31,12 +24,7 @@ def __init__(
             tasks have completed in parallel.
             spec section can't be empty. It requires at least one optional field.
         """
-        super().__init__(
-            name=name,
-            namespace=namespace,
-            client=client,
-            **kwargs,
-        )
+        super().__init__(**kwargs)
```
Comment (dbasunag) [inline ocp_resources/pipeline.py:27]: Can these resources be generated by utilizing https://github.com/RedHatQE/openshift-python-wrapper/tree/main/class_generator please?

Comment (geetikakay) [reply ocp_resources/pipeline.py:27]: oc explain doesn't show spec details . I have a bug too for that

Comment (myakove) [reply ocp_resources/pipeline.py:27]: Please try again, we do not use oc explain anymore to generate resources

Comment (geetikakay) [reply ocp_resources/pipeline.py:27]: I did give it a try :+1: 
1. I first updated the schema to get tekton resource json.

$ cat class_generator/schema/pipelinerun.json
{
  "type": "object",
  "x-kubernetes-group-version-kind": [
    {
      "group": "tekton.dev",
      "kind": "PipelineRun",
      "version": "v1beta1"
    }
  ],
  "$schema": "http://json-schema.org/schema#"

$ cat class_generator/schema/pipeline.json
{
  "type": "object",
  "x-kubernetes-group-version-kind": [
    {
      "group": "tekton.dev",
      "kind": "Pipeline",
      "version": "v1beta1"
    }
  ],
  "$schema": "http://json-schema.org/schema#"


2. I tried with working kind which is DNS in my setup just to be sure that setup is correct, it worked

3. When trying for tekton



$ class-generator --kind PipelineRun
2024-10-24T10:39:42.859879 class_generator INFO Generating class for PipelineRun
Traceback (most recent call last):
  File "/home/gkapoor/.local/bin/class-generator", line 8, in <module>
    sys.exit(main())
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/class_generator/class_generator.py", line 721, in main
    class_generator(**_kwargs)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/class_generator/class_generator.py", line 569, in class_generator
    resources = parse_explain(kind=kind)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/class_generator/class_generator.py", line 476, in parse_explain
    schema_properties: Dict[str, Any] = _kind_schema["properties"]
KeyError: 'properties'

Comment (myakove) [reply ocp_resources/pipeline.py:27]: Thanks

**Review by myakove — APPROVED** [review]

**Pull request merged**